src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Tov.cpp

```diff
@@ -79,20 +83,80 @@ class Observer {
   std::vector<double> log_enthalpy;
 };
 
+template <typename DataType>
+RelativisticEuler::Solutions::TovStar<
+    gr::Solutions::TovSolution>::RadialVariables<DataType>
+interior_solution(
+    const EquationsOfState::EquationOfState<true, 1>& equation_of_state,
+    const DataType& radius, const DataType& mass_inside_radius,
+    const DataType& log_specific_enthalpy,
+    const double log_lapse_at_outer_radius) noexcept {
+  RelativisticEuler::Solutions::TovStar<
+      gr::Solutions::TovSolution>::RadialVariables<DataType>
+      result(radius);
+  result.specific_enthalpy = Scalar<DataType>{exp(log_specific_enthalpy)};
+  result.rest_mass_density = equation_of_state.rest_mass_density_from_enthalpy(
+      result.specific_enthalpy);
+  result.pressure =
+      equation_of_state.pressure_from_density(result.rest_mass_density);
+  result.specific_internal_energy =
+      equation_of_state.specific_internal_energy_from_density(
+          result.rest_mass_density);
+  result.metric_time_potential =
+      log_lapse_at_outer_radius - log_specific_enthalpy;
+  result.dr_metric_time_potential =
+      (mass_inside_radius + 4.0 * M_PI * get(result.pressure) * cube(radius)) /
+      (radius * (radius - 2.0 * mass_inside_radius));
+  result.metric_radial_potential =
+      -0.5 * log(1.0 - 2.0 * mass_inside_radius / radius);
+  result.dr_metric_radial_potential =
+      (4.0 * M_PI * radius *
+           (get(result.specific_enthalpy) * get(result.rest_mass_density) -
+            get(result.pressure)) -
+       mass_inside_radius / square(radius)) /
+      (1.0 - 2.0 * mass_inside_radius / radius);
+  result.metric_angular_potential = make_with_value<DataType>(radius, 0.0);
+  result.dr_metric_angular_potential = make_with_value<DataType>(radius, 0.0);
+  return result;
+}
+
+template <typename DataType>
+RelativisticEuler::Solutions::TovStar<
+    gr::Solutions::TovSolution>::RadialVariables<DataType>
+vacuum_solution(const DataType& radius, const double total_mass) noexcept {
+  RelativisticEuler::Solutions::TovStar<
+      gr::Solutions::TovSolution>::RadialVariables<DataType>
+      result(radius);
+  result.specific_enthalpy = make_with_value<Scalar<DataType>>(radius, 1.0);
+  result.rest_mass_density = make_with_value<Scalar<DataType>>(radius, 0.0);
+  result.pressure = make_with_value<Scalar<DataType>>(radius, 0.0);
+  result.specific_internal_energy =
+      make_with_value<Scalar<DataType>>(radius, 0.0);
+  result.metric_time_potential = 0.5 * log(1.0 - 2.0 * total_mass / radius);
```

Comment:
I assume the reason `2.0 * total_mass / radius` is not being precomputed is a detail of radius possibly being a DataVector?